sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1286,6 +1341,360 @@ __ESIMD_API void lsc_store2d(T *Ptr, unsigned SurfaceWidth,
       Vals.data());
 }
 
+/// <summary>
+///  Container class to hold parameters for \c load2d/store2d \c functions
+/// </summary>
+/// @tparam T Type of data to load/store
+/// @tparam BlockWidth the block width in number of elements
+/// @tparam BlockHeight block height in number of elements
+/// @tparam NBlocks Number of blocks
+template <typename T, int BlockWidth, int BlockHeight, int NBlocks>
+class config_2d_mem_access {
```

Comment:
(minor nit) Maybe the naming can be more consistent: `config_2d_mem_access` (with `2d` a separate word) vs. `load2d`/`store2d`

Reply:
I like lsc_load_2d(), lsc_store_2d() and config_2d_mem_access() more than the current variant without underscore before "2d". 
The problem is that the old method is already named without underscore, i.e. it is lsc_load2d.
Using new naming would cause us to create another variant for older method, and deprecating lsc_load2d(), i.e.:
```
template <...>
__ESIMD_NS::simd<T, N>
lsc_load_2d(const T *Ptr, unsigned SurfaceWidth, unsigned SurfaceHeight,
           unsigned SurfacePitch, int X, int Y) {
  <implementation here>
}

template <...>
__SYCL_DEPRECATED("use lsc_load_2d()") __ESIMD_NS::simd<T, N>
lsc_load2d(const T *Ptr, unsigned SurfaceWidth, unsigned SurfaceHeight,
           unsigned SurfacePitch, int X, int Y) {
  return lsc_load_2d<...>(...);
}
```
Not sure if that worth it.

Reply:
It may be a good idea, actually and it is simpler to do that now, while everything is in experimental namespace.
It will a good correct naming precedent, and will make it easier later to move structs/methods out of experimental namespace as well.

Reply:
Thank you! Please address this comment asking to add deprecated warning: https://github.com/intel/llvm/commit/87fe557be88ce4320d01de848141c12105c3c5e4#r98977535